Withdraw channel payment flow

drillprop opened the pull request:
Fix #3054

Designs: 
https://www.figma.com/file/ZFRXsyGvnBXnkDUUGc5vyR/My-payments?node-id=1602%3A25902
https://www.figma.com/file/uHzbhNX9dP4QtZbIapvZI6/Transfer-dialog?node-id=336%3A2174

drillprop committed:
fix label issue

drillprop committed:
fix max button, add max button to withdraw dialog

drillprop committed:
improve validation

drillprop committed:
bring back dialog in member dropdowns

rafalpawlow reviewed (changes requested):
`Take fees into account when user clicks "Max" button`

When I click `MAX` I got 7 in input but according to this AC, it should be 6.94. Also it's not properly validated, I'm able to send the transaction from the video below.

https://user-images.githubusercontent.com/13706246/186150512-c3bc5d00-df37-4bfc-9b21-7c624fcf4182.mov

rafalpawlow commented:
Updated video with failed transaction

https://user-images.githubusercontent.com/13706246/186152030-da8ad2f9-5854-4da3-99c6-7637e82415a6.mov

rafalpawlow reviewed (changes requested):
When I have 0 balance on destination account, nothing happens when I click `MAX`

https://user-images.githubusercontent.com/13706246/186155792-745ec7de-1b1c-409a-802d-90e337a13173.mov

drillprop committed:
substract existentialDeposit

drillprop committed:
cr fixes

drillprop committed:
improvements

drillprop committed:
compare with rounded balance

drillprop committed:
revert previous commit

This reverts commit 981db193736d01f6d84511d6f3b873ba59235433.

drillprop commented:
Ok, did some improvements: 
- Based on the slack discussion I subtracted existential deposit from the **channel balance**. So basically, in the member dropdown, you'll see only available tokens. I also added a small comment, that this will change in the future and we will use the channel state bloat bond.  
- We don't subtract fees when we click max in send funds dialogs. When you click max input is fulfilled with all the balance that you have. This was consulted with Adam. We added a small description in the input form field, that the transaction fee will be deducted from the amount in the input. And we will do this under the hood when the user clicks max. 
- When we click the max button, the value in the input is rounded **down** to the two decimal places. This causes some confusion, but I have no idea how to do this better. For example, I have a channel balance that equals 49,4998333334 tokens(494998333334 hapi). In the member dropdown, this will be displayed as 49.5 because NumberFormat for some reason is rounding this value up. If I click the max button the value in the input will equal to `49,49` This creates some inconsistency. Check the video:

https://user-images.githubusercontent.com/51168865/186354935-3ebf4639-8a92-4ecf-bec1-456e998259a5.mov

I could round the input value up as we did in the NumberFormat, but this value will be incorrect, because you don't have that amount of tokens, and validation will fail. So I decided to leave this as I implemented. Feel free if you have a better solution to do this. 

Also, there is another problem If I have `49,4998333334` and send `49,49` there will be always some hapi that's left in the wallet - You'll have `0,0098333334`. So basically you can't withdraw all the money that you have, there will be always some pennies left.

kdembler commented on packages/atlas/src/components/_inputs/FormField/FormField.tsx:
```diff
@@ -99,22 +99,19 @@ export const FormField = memo(
                     </Text>
                   )}
                   {tooltip && <StyledInformation {...tooltip} />}
+                  {headerNode && headerNode}
```
This is equivalent to just `headerNode`. If you want to be safe from `0` for example, you would need to do `headerNode ? headerNode : null`

kdembler commented on packages/atlas/src/providers/joystream/joystream.hooks.ts:
```diff
@@ -155,15 +161,31 @@ export const useSubscribeAccountBalance = (controllerAccount?: string | null) =>
+            const rewardBalance = new BN(balance).sub(chainState.existentialDeposit)
+            setAccountBalance(rewardBalance)
+            return
+          }
```
I think first you need to confirm that the account actually has any funds. Because (at least currently, without the channel state bloat bond being in the account) new channels will have initial balance of 0 (the account doesn't really "exist" until it gets it first income). So for new channels the balance we display will actually be negative (0 - existential deposit):
<img width="274" alt="CleanShot 2022-08-25 at 16 26 36@2x" src="https://user-images.githubusercontent.com/12646744/186692455-888a04aa-6699-4f2d-a98f-014db483e5f3.png">

(-0 because of rounding most likely)

drillprop committed:
cr fixes

drillprop committed:
isRewardAccount flag

drillprop committed:
fix issue with infinitely loading reward account balance

drillprop committed:
fix issue with not loading balance
